libusb/os/linux_usbfs.c

```diff
@@ -1407,6 +1408,129 @@ static int op_wrap_sys_device(struct libusb_context *ctx,
 	return r;
 }
 
+static int check_subsystem(const char *sys_path, const char* subsystem)
+{
+	char *path, *subsystem_path;
+	int ret;
+
+	ret = asprintf(&path, "%s/subsystem", sys_path);
+	if (ret < 0)
+		return LIBUSB_ERROR_NO_MEM;
+
+	subsystem_path = realpath(path, NULL);
+	free(path);
+	if (!subsystem_path && errno == ENOMEM)
+		return LIBUSB_ERROR_NO_MEM;
```

Comment:
If `subsystem_path` is NULL but errno is something else, shouldn't you be returning a libusb error here?

Reply:
I don't think you addressed this one...

Reply:
It should be `errno != ENOENT` as we check all paths even if there is no `./subsystem` directory which would mean it doesn't have a subsystem.

Reply:
If (somehow!) `realpath` returns NULL *and* sets `errno=ENOENT`, you will return the result of `asprintf()` (the number of characters printed). Is that what we want?

Reply:
no, but happened to work as a non-match would be a positive number.